components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt

```diff
@@ -514,6 +591,39 @@ class SitePermissionsFeature(
     private fun PermissionRequest.isForAutoplay() =
         this.permissions.any { it is ContentAutoPlayInaudible || it is ContentAutoPlayAudible }
 
+    private fun PermissionRequest.isForNotification() =
+        this.permissions.any { it is ContentNotification }
+
+    private fun PermissionRequest.isForCamera() =
+        this.permissions.any {
+            it is ContentVideoCamera || it is ContentVideoCapture || it is AppCamera
+        }
+
+    private fun PermissionRequest.isForAutoplayInaudible() =
+        this.permissions.any { it is ContentAutoPlayInaudible }
+
+    private fun PermissionRequest.isForAutoplayAudible() =
+        this.permissions.any { it is ContentAutoPlayAudible }
+
+    private fun PermissionRequest.isForLocation() =
+        this.permissions.any {
+            it is ContentGeoLocation ||
+                it is AppLocationCoarse ||
+                it is AppLocationFine
+        }
+
+    private fun PermissionRequest.isForMicroPhone() =
```

Comment:
`Microphone`? Except if we are talking about very small phones here :D

Reply:
XD